src/coreComponents/constitutive/fluid/MultiFluidBase.hpp

```diff
@@ -636,49 +581,41 @@ void
 MultiFluidBase::KernelWrapper::
   convertToMassFractions( real64 const (&dCompMoleFrac_dCompMassFrac)[maxNumComp][maxNumComp],
                           real64 const (&phaseMolecularWeight)[maxNumPhase],
-                          real64 const (&dPhaseMolecularWeight_dPres)[maxNumPhase],
-                          real64 const (&dPhaseMolecularWeight_dTemp)[maxNumPhase],
-                          real64 const (&dPhaseMolecularWeight_dGlobalCompFrac)[maxNumPhase][maxNumComp],
+                          real64 const (&dPhaseMolecularWeight)[maxNumPhase][maxNumComp+2],
                           PhaseProp::SliceType const phaseFrac,
                           PhaseComp::SliceType const phaseCompFrac,
-                          arraySlice2d< real64, multifluid::USD_PHASE_DC - 2 > const dPhaseDens_dGlobalCompFrac,
-                          arraySlice2d< real64, multifluid::USD_PHASE_DC - 2 > const dPhaseVisc_dGlobalCompFrac ) const
+                          arraySlice2d< real64, multifluid::USD_PHASE_DC - 2 > const dPhaseDens,
+                          arraySlice2d< real64, multifluid::USD_PHASE_DC - 2 > const dPhaseVisc ) const
 {
   convertToPhaseMassFractions( phaseMolecularWeight,
-                               dPhaseMolecularWeight_dPres,
-                               dPhaseMolecularWeight_dTemp,
-                               dPhaseMolecularWeight_dGlobalCompFrac,
+                               dPhaseMolecularWeight,
                                phaseFrac );
   convertToPhaseCompMassFractions( phaseMolecularWeight,
-                                   dPhaseMolecularWeight_dPres,
-                                   dPhaseMolecularWeight_dTemp,
-                                   dPhaseMolecularWeight_dGlobalCompFrac,
+                                   dPhaseMolecularWeight,
                                    phaseFrac,
                                    phaseCompFrac );
   computeDerivativesWrtMassFractions( dCompMoleFrac_dCompMassFrac,
                                       phaseFrac,
                                       phaseCompFrac,
-                                      dPhaseDens_dGlobalCompFrac,
-                                      dPhaseVisc_dGlobalCompFrac );
+                                      dPhaseDens,
+                                      dPhaseVisc );
 }
 
-template< integer maxNumComp, integer maxNumPhase >
+template< integer maxNumDof, integer maxNumPhase >
```

Comment:
Here and after, I would definitely `doxygen` the functions.
Please do not forget the `@tparam maxNumDof` and specially which `dof` we are talking about.

Reply:
I added the doxygen above in the class where the function is declared, with some information about which dof (pressure, temperature, and component fractions) we are talking about.